controllers/gslb_controller.go

```diff
@@ -195,13 +198,78 @@ func (r *GslbReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			return nil
 		})
 
+	createGslbFromIngress := func(annotationKey string, annotationValue string, a handler.MapObject, strategy string) {
+		log.Info(fmt.Sprintf("Detected strategy annotation(%s:%s) on Ingress(%s)",
+			annotationKey, annotationValue, a.Meta.GetName()))
+		c := mgr.GetClient()
+		ingressToReuse := &v1beta1.Ingress{}
+		err := c.Get(context.Background(), client.ObjectKey{
+			Namespace: a.Meta.GetNamespace(),
+			Name:      a.Meta.GetName(),
+		}, ingressToReuse)
+		if err != nil {
+			log.Info(fmt.Sprintf("Ingress(%s) does not exist anymore. Skipping...", a.Meta.GetName()))
+			return
+		}
+		gslbExist := &k8gbv1beta1.Gslb{}
+		err = c.Get(context.Background(), client.ObjectKey{
+			Namespace: a.Meta.GetNamespace(),
+			Name:      a.Meta.GetName(),
+		}, gslbExist)
+		if err == nil {
+			log.Info(fmt.Sprintf("Gslb(%s) already exists. Skipping...", gslbExist.Name))
+			return
+		}
+		gslb := &k8gbv1beta1.Gslb{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:   a.Meta.GetNamespace(),
+				Name:        a.Meta.GetName(),
+				Annotations: a.Meta.GetAnnotations(),
+			},
+			Spec: k8gbv1beta1.GslbSpec{
+				Ingress: ingressToReuse.Spec,
+				Strategy: k8gbv1beta1.Strategy{
+					Type: strategy,
+				},
+			},
+		}
+
+		if strategy == "failover" {
+			for annotationKey, annotationValue := range a.Meta.GetAnnotations() {
+				if annotationKey == "k8gb.io/primarygeotag" {
```

Comment:
What happens in the case where `k8gb.io/primarygeotag` annotation is missing? Should we log an error and ignore?

Reply:
@donovanmuller good catch, thanks a lot. Implemented in https://github.com/AbsaOSS/k8gb/pull/205/commits/7fb78f6a5358f680a262584c5d8bb5b89f5b5899